src/app/services/api.service.ts

```diff
@@ -1,9 +1,48 @@
 import { Injectable } from '@angular/core';
+import { CardComponent} from "../card/card.component";
+
 
 @Injectable({
   providedIn: 'root'
 })
 export class ApiService {
 
-  constructor() { }
+  url = "http://localhost:3000/tasks";
+  catTipsUrl = "https://catfact.ninja/fact";
+
+  async getAllTasks(): Promise<CardComponent[]> {
+    const data = await fetch(this.url);
+    return await data.json() ?? [];
+  }
+
+  async getTaskById(id:number): Promise<CardComponent | undefined> {
+    const data = await fetch(`${this.url}/${id}`);
+    return await data.json() ?? {};
+  }
+
+  async createTask(data:CardComponent): Promise<void> {
+    await fetch(this.url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(data)
+    })
+  }
+
+  async deleteTask(id:number): Promise<void> {
+    await fetch(`${this.url}/${id}`, {
+      method: 'DELETE',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+    })
+  }
+
+  async getCatTip(): Promise<any> {
```

Comment:
This method should return a `Promise<string>`,  try parsing the data from the response, i think the object property was `.fact`, you can check it in the TA we did in class, and return that string value or an empty string if no fact is returned